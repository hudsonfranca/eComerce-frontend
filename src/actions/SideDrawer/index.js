export const mobileCategories = () => ({
  type: "mobileCategories"
});

export const mobileAccount = () => ({
  type: "mobileAccount"
});

export const sideDrawerClose = () => ({
  type: "close"
});
