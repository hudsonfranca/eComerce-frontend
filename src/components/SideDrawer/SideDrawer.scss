@import "../style/scss/variables";
@import "../style/scss/HamburguerButton";
@import "../style/scss/media";

.sideDrawer {
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
  background-color: $primaryColor;
  position: absolute;
  z-index: 200;
  transform: translateX(-100%);
  transition: transform 0.5s cubic-bezier(0.55, 0.055, 0.675, 0.19);

  &.open {
    transform: translateX(0);
  }

  & > button {
    align-self: flex-end;
  }

  .sideDrawer-content {
    flex: 1;
    @include for-tablet-portrait-up {
      width: 75%;
      align-self: center;
    }
  }
}
