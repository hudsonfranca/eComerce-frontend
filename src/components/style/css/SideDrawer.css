.hamburguerButton {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
      -ms-flex-pack: space-evenly;
          justify-content: space-evenly;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  width: 40px;
  height: 40px;
  border-radius: 5px;
  background: #f9f6f7;
  cursor: pointer;
  -webkit-transition: 0.5s;
  transition: 0.5s;
}

.hamburguerButton:hover, .hamburguerButton:active {
  background-color: white;
}

.hamburguerButton .rbline {
  width: 30px;
  height: 3px;
  background-color: #313131;
  border-radius: 5px;
}

.hamburguerButton.close {
  -webkit-transform: rotate(180deg);
          transform: rotate(180deg);
}

.hamburguerButton.close .rbline:nth-child(1) {
  -webkit-transform: rotate(45deg) translate(8px, 8px);
          transform: rotate(45deg) translate(8px, 8px);
}

.hamburguerButton.close .rbline:nth-child(2) {
  opacity: 0;
}

.hamburguerButton.close .rbline:nth-child(3) {
  -webkit-transform: rotate(-45deg) translate(6px, -7px);
          transform: rotate(-45deg) translate(6px, -7px);
}

.sideDrawer {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  width: 100%;
  height: 100%;
  background-color: #313131;
  position: absolute;
  z-index: 200;
  -webkit-transform: translateX(-100%);
          transform: translateX(-100%);
  -webkit-transition: -webkit-transform 0.5s cubic-bezier(0.55, 0.055, 0.675, 0.19);
  transition: -webkit-transform 0.5s cubic-bezier(0.55, 0.055, 0.675, 0.19);
  transition: transform 0.5s cubic-bezier(0.55, 0.055, 0.675, 0.19);
  transition: transform 0.5s cubic-bezier(0.55, 0.055, 0.675, 0.19), -webkit-transform 0.5s cubic-bezier(0.55, 0.055, 0.675, 0.19);
}

.sideDrawer.open {
  -webkit-transform: translateX(0);
          transform: translateX(0);
}

.sideDrawer > button {
  -ms-flex-item-align: end;
      align-self: flex-end;
}

.sideDrawer .sideDrawer-content {
  -webkit-box-flex: 1;
      -ms-flex: 1;
          flex: 1;
}

@media (min-width: 600px) {
  .sideDrawer .sideDrawer-content {
    width: 75%;
    -ms-flex-item-align: center;
        -ms-grid-row-align: center;
        align-self: center;
  }
}
/*# sourceMappingURL=SideDrawer.css.map */