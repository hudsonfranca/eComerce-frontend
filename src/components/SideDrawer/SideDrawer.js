import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { sideDrawerClose } from "../../actions/SideDrawer";
import "../style/css/SideDrawer.css";

export function SideDrawer({ isOpen, sideDrawerClose, content }) {
  return (
    <div className={`sideDrawer ${isOpen}`} data-testid="SideDrawer">
      <button
        type="button"
        className="hamburguerButton close"
        onClick={sideDrawerClose}
      >
        <div className="rbline" />
        <div className="rbline" />
        <div className="rbline" />
      </button>
      <div className="sideDrawer-content">{content}</div>
    </div>
  );
}

const mapStateToProps = state => ({
  isOpen: state.sideDrawerState.isOpen,
  content: state.sideDrawerState.content
});

const mapDispatchToProps = dispatch => ({
  sideDrawerClose: () => dispatch(sideDrawerClose())
});

export default connect(mapStateToProps, mapDispatchToProps)(SideDrawer);
