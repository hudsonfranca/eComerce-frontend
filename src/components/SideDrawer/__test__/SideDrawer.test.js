import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import { SideDrawer } from "../SideDrawer";

afterEach(cleanup);

it(" SideDrawer renders without crashing", () => {
  const div = document.createElement("div");

  ReactDOM.render(<SideDrawer />, div);
});

it("SideDrawer matches snapshot", () => {
  const tree = renderer.create(<SideDrawer />).toJSON();

  expect(tree).toMatchSnapshot();
});
