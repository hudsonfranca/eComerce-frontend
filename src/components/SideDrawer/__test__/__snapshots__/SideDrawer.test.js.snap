// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideDrawer matches snapshot 1`] = `
<div
  className="sideDrawer undefined"
  data-testid="SideDrawer"
>
  <button
    className="hamburguerButton close"
    type="button"
  >
    <div
      className="rbline"
    />
    <div
      className="rbline"
    />
    <div
      className="rbline"
    />
  </button>
  <div
    className="sideDrawer-content"
  />
</div>
`;

exports[`matches snapshot 1`] = `
<div
  className="sideDrawer undefined"
  data-testid="SideDrawer"
>
  <button
    className="hamburguerButton close"
    type="button"
  >
    <div
      className="rbline"
    />
    <div
      className="rbline"
    />
    <div
      className="rbline"
    />
  </button>
  <div
    className="sideDrawer-content"
  />
</div>
`;
